packages/dashboard-core-plugins/src/panels/MockFileStorageTable.ts

```diff
@@ -25,6 +29,25 @@ export class MockFileStorageTable implements FileStorageTable {
   }
 
   setExpanded(path: string, expanded: boolean): Promise<void> {
+    // only sets expansion for the FileExplorer in NewItemModal
+    if (path.charAt(path.length - 1) === '/') {
+      for (let i = 0; i < this.items.length; i += 1) {
+        const item = this.items[i];
+        if (
+          isDirectory(item) &&
+          (`${item.filename}/` === path || item.filename === path)
+        ) {
+          item.isExpanded = expanded;
+        }
+      }
+    }
+    return Promise.resolve();
+  }
+
+  collapseAll(): Promise<void> {
+    for (let i = 0; i < this.items.length; i += 1) {
+      this.setExpanded(`${this.items[i].filename}/`, false);
+    }
```

Comment:
We're iterating through all items here, and then iterating through them again in `setExpanded`, which is `O(N^2)`. To be more efficient, we should only need to loop through them once:
```suggestion
    for (let i = 0; i < this.items.length; i += 1) {
      const item = this.items[i];
      if (isDirectory(item)) {
        item.isExpanded = false;
      }
    }
```